builtin/common/misc_helpers.lua

```diff
@@ -198,9 +198,44 @@ function table.indexof(list, val)
 	return -1
 end
 
+table.indexof = table.index_of
 assert(table.indexof({"foo", "bar"}, "foo") == 1)
 assert(table.indexof({"foo", "bar"}, "baz") == -1)
 
+--------------------------------------------------------------------------------
+function table.count(t)
+	local i = 0
+	for _ in pairs(t) do
+		i = i + 1
+	end
+	return i
+end
+
+--------------------------------------------------------------------------------
+function table.random(t)
+	return t[math.random(#t)]
+end
+
+--------------------------------------------------------------------------------
+function table.map(t, f)
+	local ret = {}
+	for key, value in pairs(t) do
+		ret[key] = f(value)
+	end
+	return ret
+end
+
+--------------------------------------------------------------------------------
+function table.map_inplace(t, f)
+	for key, value in pairs(t) do
+		value = f(value)
+		if value ~= nil then
```

Comment:
this behaviour does not match what `.map` does, it's not documented either

Reply:
that was a mistake